### PR TITLE
Add kustomization sync

### PR DESCRIPTION
**Pull request by jpellizzari**
Adds a button to Sync a kustomization. Copied from the flux2 implementation.

### CODE REVIEW

File: pkg/clustersserver/clustersserver.go [hunk 1]
```diff
@@ -158,3 +165,112 @@ func (s *Server) ListSources(ctx context.Context, msg *pb.ListSourcesReq) (*pb.L
...
+		return err
+	}
+
+	if bucket.Annotations == nil {
+		bucket.Annotations = map[string]string{
+			meta.ReconcileAtAnnotation: time.Now().Format(time.RFC3339Nano),
+		}
+	} else {
+		bucket.Annotations[meta.ReconcileAtAnnotation] = time.Now().Format(time.RFC3339Nano)
+	}
```
Comment (hiddeco) [inline pkg/clustersserver/clustersserver.go:210]:
```suggestion
	if bucket.Annotations == nil {
		bucket.Annotations = map[string]string{}
	}
	bucket.Annotations[meta.ReconcileAtAnnotation] = time.Now().Format(time.RFC3339Nano)
```

File: pkg/clustersserver/clustersserver.go [hunk 2]
```diff
@@ -158,3 +165,112 @@ func (s *Server) ListSources(ctx context.Context, msg *pb.ListSourcesReq) (*pb.L
...
+		return err
+	}
+
+	if repository.Annotations == nil {
+		repository.Annotations = map[string]string{
+			meta.ReconcileAtAnnotation: time.Now().Format(time.RFC3339Nano),
+		}
+	} else {
+		repository.Annotations[meta.ReconcileAtAnnotation] = time.Now().Format(time.RFC3339Nano)
+	}
```
Comment (hiddeco) [inline pkg/clustersserver/clustersserver.go:187]:
```suggestion
	if repository.Annotations == nil {
		repository.Annotations = map[string]string{}
	}
	repository.Annotations[meta.ReconcileAtAnnotation] = time.Now().Format(time.RFC3339Nano)
```

**Review by blaquezzz — APPROVED** [review]

**Review by squaremo — COMMENTED** [review]
I feel qualified only to comment ..

File: ui/lib/rpc/clusters.ts [hunk 3]
```diff
@@ -262,13 +262,61 @@ const JSONToListSourcesRes = (m: ListSourcesRes | ListSourcesResJSON): ListSourc
...
+interface SyncKustomizationReqJSON {
+    contextName: string;
+    kustomizationName: string;
+    kustomizationNamespace: string;
+    withSource: boolean;
+    
+}
+
+
+const SyncKustomizationReqToJSON = (m: SyncKustomizationReq): SyncKustomizationReqJSON => {
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/lib/rpc/clusters.ts:282]: This is a generated file

File: ui/pages/KustomizationDetail.tsx [hunk 4]
```diff
@@ -13,11 +16,30 @@ type Props = {
...
+    const clusters = new DefaultClusters("/api/clusters", wrappedFetch);
+
+    setSyncing(true);
+
+    clusters
+      .syncKustomization({
+        contextname: currentContext,
+        withsource: false,
+        kustomizationname: kustomizationId,
+        kustomizationnamespace: "flux-system",
```
[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/pages/KustomizationDetail.tsx:36]: Currently yes, but a good note. I don't fully understand what role namespaces place with regard to Kustomizations. Should the kustomization be in the namespace of the deployment that it create/refers to?

[the remark this answers is not in the document]
Comment (stefanprodan) [reply ui/pages/KustomizationDetail.tsx:36]: We need a namespace filter like in Kubernetes dashboard, by default it should select all namespaces. Flux objects can be created in any namespace.

[the remark this answers is not in the document]
Comment (jpellizzari) [reply ui/pages/KustomizationDetail.tsx:36]: Since this is hard-coded only in the client currently, I will address this in a separate PR.

**Comment by jpellizzari** [comment]
@squaremo Consolidated the reconciliation stuff: https://github.com/fluxcd/webui/pull/8/commits/b65b8500722d1b5d31b00822285f5b8448622b2d

**Review by squaremo — APPROVED** [review]
The Go/K8s client part of this looks fine to me. I think it'd be possible to factor out the reconciliation still further -- the kustomization object is _also_ a runtime.Object. But it might be better for us to think longer term about making the abstractions from the CLI available in a module, for this purpose.

**Pull request merged**